### DISCUSSION TITLE
Using .onnx file outside MMDeploy

### DISCUSSION THREAD

**Discussion by dnth** [open-mmlab/mmdeploy#213] [Q&A] [answered]
Can the exported models e.g. the .onnx file be used for inference outside of MMDeploy without using the MMDeploy SDK? I wish to use the .onnx file in my own inference code such as [this](https://github.com/microsoft/onnxruntime-inference-examples/blob/main/python/OpenVINO_EP/yolov4_object_detection/yolov4.py)

**Comment by grimoire** [comment, marked as answer]
If you mean you want to export the model we supported and use it in your own project. YES, of cause. Just use it with the apis provided by the backend engine and it will have no difference.
If you want to export your custom model (which does not come from OpenMMLab), Errrr, I am not sure. You might have to do some modifications to the code so it can be traced. We have provided a tool called function rewriter, you can use it to ease the process, there should be a doc about how to write the rewrite function.
